src/transformStateWithClones.js

```diff
@@ -6,8 +6,31 @@
  *
  * @return {Object[]}
  */
+
 function transformStateWithClones(state, actions) {
-  // write code here
+  const arrState = [];
```

Comment:
Consider initializing `arrState` with the initial state to avoid checking if `arrState` is empty on each iteration. This can simplify the logic and potentially improve performance.